### PR TITLE
Added params-info howto

### PR DESCRIPTION
**Pull request by rolandgvc**
Added HOWTO for logging number of trainable parameters and their memory footprint.

### CODE REVIEW

**Comment by rolandgvc** [comment]
@googlebot I signed it!

**Comment by rolandgvc** [comment]
@avital are these ok as a separate function or would you like them written directly in `train()` ?

**Comment by avital** [comment]
Thanks! It's good that you made it a separate function, but it doesn't look like it's called from anywhere?

**Comment by rolandgvc** [comment]
Wasn't sure if calling the function (given it has no arguments) should be included in the howto vs just writing those lines in the train function directly

**Comment by avital** [comment]
Oh the HOWTO should apply to the base example in a way that 100% produces the described behavior. It's a complete guide, like "do exactly X in your code and you can achieve Y"

File: howtos/diffs/params-info.diff [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+--- a/examples/mnist/train.py
++++ b/examples/mnist/train.py
+@@ -174,6 +174,17 @@ def get_datasets():
+   return train_ds, test_ds
+ 
+ 
++def params_info(model):
++  """ Log number of trainable parameters and their memory footprint"""
++  logging.info('Number of trainable paramters: %d', 
++               onp.sum([onp.prod(layer_params) for layer_params in jax.tree_map(
```
[the remark this answers is not in the document]
Comment (rolandgvc) [reply howtos/diffs/params-info.diff:12]: Thanks for the suggestions! The [0] is the same as `tree_leaves` (as we've discussed) which I will be using from now on.

**Comment by rolandgvc** [comment]
not sure why the apply error fails in the build, I've tested it locally and has no problems

**Comment by avital** [comment]
Can you please resolve whatever comments you think are addressed? I'm not sure what the current state is.

**Comment by avital** [comment]
@danielsuo do you have any idea why this build is failing? @rolandgvc says applying the HOWTO diff works for him locally. (though I haven't tested it myself)

**Comment by danielsuo** [comment]
You probably need to rebase to master. Github first merges your code into master (see logs below)

```bash
commit e6d756aabd655b65910c5b8aa106dae3d847ebca
Author: Roland Gavrilescu <gavrilescu.roland@gmail.com>
Date:   Fri May 29 14:45:51 2020 +0200

    Merge 45d0199a5291280ad2d7c24c70b5cac79c937e2e into 967103ee5f7414ec1d543ea9f9aeab1543fc5807
```

**Comment by danielsuo** [comment]
Perhaps for a separate issue since @joaogui1 got tripped up with this in #267 as well, but I think we may want to continue enforcing that `howto`s work on master. Otherwise, broken `howto`s can slip into master and we only discover once it starts holding up future PRs (where the author may not be running `check_howto_diffs.sh` in their testing).

A little painful for the `howto` writer, but hopefully a minor inconvenience once base examples (e.g., `mnist`) settle down.

**Comment by danielsuo** [comment]
@rolandgvc I think you'll need to clean up the history since you've imported a lot of commits not relevant to this PR. In this case, less for reasons of hygiene than the `cla/google` check looks like it's failing because one of the commit authors you imported hasn't signed the CLA.

**Comment by marcvanzee** [comment]
Hi @rolandgvc, I'm seeing this PR for the first time and I'd like to review it. However I think something went wrong with rebasing, since as @danielsuo says as well, it seems many files have changed unrelated to your change. Could you pleas e rebase so I can review your PR? Thanks!

**Pull request closed without merging**